_posts/2020-08-05-Mocks-and-Spies-in-Minitest.md

```diff
@@ -0,0 +1,78 @@
+---
+layout: post
+title: "Mocks & Spies in Minitest"
+author: podrezo
+date: 2020-08-05
+excerpt: "For one of our smaller ruby projects we didn't want to bring in all of rspec, so we decided to go with minitest as our unit test runner and assertion library. Doing mocks and spies in minitest proved a little bit challenging to get the hang of initially but it's actually quite easy. This article will explain how to do it."
+---
+For one of our smaller ruby projects we didn't want to bring in all of rspec, so we decided to go with minitest as our unit test runner and assertion library. Doing mocks and spies in minitest proved a little bit challenging to get the hang of initially but it's actually quite easy. This article will explain how to do it.
+
+First, let's consider two classes - one helper class which we will be mocking out, and the other is the class under test.
+
+
+```ruby
+class MyHelperClass
+  def upcase_instance(value)
+    # value.upcase
+    raise "Fail the test if the real thing is called"
+  end
+
+  def self.upcase_static(value)
+    # value.upcase
+    raise "Fail the test if the real thing is called"
+  end
+end
+
+class MyClassUnderTest
+  def use_upcase_instance(value)
+    helper_class = MyHelperClass.new
+    helper_class.upcase_instance(value)
+  end
+
+  def use_upcase_static(value)
+    MyHelperClass.upcase_static(value)
+  end
+end
+```
+
+Note that if the real helper class's methods are called the test will fail to illustrate that we're doing the correct thing. In your test spec file you wil need to bring in the necessary `require`s for minitest:
```

Comment:
I'd use `In you test file (...)`. The `spec` is not adding anything unless you're trying to say that it is a test file and a specification file at the same time.
When I worked with Minitest I don't remember people referencing the files as spec files